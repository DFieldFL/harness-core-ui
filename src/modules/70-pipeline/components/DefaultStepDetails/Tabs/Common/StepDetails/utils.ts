export function encodeURIWithReservedChars(uri: string): string {
  return encodeURIComponent(uri).replace(/[;,/?:@&=+$#]/g, function (c) {
    return '%' + c.charCodeAt(0).toString(16)
  })
}
