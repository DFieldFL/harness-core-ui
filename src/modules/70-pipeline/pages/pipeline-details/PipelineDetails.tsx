import React from 'react'
import { Container, Layout, Icon, Color } from '@wings-software/uikit'
import { NavLink, useParams } from 'react-router-dom'
import { Page } from '@common/exports'
import routes from '@common/RouteDefinitions'
import { useGetPipelineSummary } from 'services/cd-ng'
import { Breadcrumbs } from '@common/components/Breadcrumbs/Breadcrumbs'
import { useAppStore, useStrings } from 'framework/exports'
import type { PipelinePathProps } from '@common/interfaces/RouteInterfaces'
import css from './PipelineDetails.module.scss'

export default function PipelineDetails({ children }: React.PropsWithChildren<{}>): React.ReactElement {
  const { orgIdentifier, projectIdentifier, pipelineIdentifier, accountId } = useParams<PipelinePathProps>()
  const { data: pipeline } = useGetPipelineSummary({
    pipelineIdentifier,
    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier }
  })
  const { projects } = useAppStore()
  const project = projects.find(({ identifier }) => identifier === projectIdentifier)
  const { getString } = useStrings()

  return (
    <>
      <Page.Header
        title={
          <Layout.Vertical spacing="xsmall">
            <Breadcrumbs
              links={[
                {
                  url: routes.toCDProjectOverview({ orgIdentifier, projectIdentifier, accountId }),
                  label: project?.name as string
                },
                {
                  url: routes.toCDPipelines({
                    orgIdentifier,
                    projectIdentifier,
                    accountId
                  }),
                  label: getString('pipelines')
                },
                { url: '#', label: pipeline?.data?.name || '' }
              ]}
            />
          </Layout.Vertical>
        }
        toolbar={
          <Container>
            <Layout.Horizontal spacing="medium">
              <NavLink
                className={css.tags}
                activeClassName={css.activeTag}
                to={routes.toCDPipelineDeploymentList({
                  orgIdentifier,
                  projectIdentifier,
                  pipelineIdentifier,
                  accountId
                })}
              >
                {getString('executionsText')}
              </NavLink>

              <NavLink
                className={css.tags}
                activeClassName={css.activeTag}
                to={routes.toCDInputSetList({ orgIdentifier, projectIdentifier, pipelineIdentifier, accountId })}
              >
                {getString('inputSetsText')}
              </NavLink>
              <NavLink
                className={css.tags}
                activeClassName={css.activeTag}
                to={routes.toCDTriggersPage({ orgIdentifier, projectIdentifier, pipelineIdentifier, accountId })}
              >
                {getString('pipeline-triggers.triggersLabel')}
              </NavLink>

              <NavLink
                className={css.tags}
                to={routes.toCDPipelineStudio({ orgIdentifier, projectIdentifier, pipelineIdentifier, accountId })}
              >
                <Icon name="pipeline-ng" size={20} style={{ marginRight: '8px' }} color={Color.BLUE_600} />
                {getString('studioText')}
              </NavLink>
            </Layout.Horizontal>
          </Container>
        }
      />
      <Page.Body>{children}</Page.Body>
    </>
  )
}
