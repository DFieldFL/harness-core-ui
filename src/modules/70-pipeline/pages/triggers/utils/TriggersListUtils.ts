import type { IconName } from '@wings-software/uicore'
// temporary mock data
import type { AddDrawerMapInterface } from '@common/components/AddDrawer/AddDrawer'
import type { GetActionsListQueryParams } from 'services/pipeline-ng'
import type { StringKeys } from 'framework/strings'
import { TriggerTypes } from './TriggersWizardPageUtils'

export const GitSourceProviders: Record<
  GetActionsListQueryParams['sourceRepo'],
  { value: GetActionsListQueryParams['sourceRepo']; iconName: IconName }
> = {
  GITHUB: { value: 'GITHUB', iconName: 'github' },
  GITLAB: { value: 'GITLAB', iconName: 'service-gotlab' },
  BITBUCKET: { value: 'BITBUCKET', iconName: 'bitbucket' },
  AWS_CODECOMMIT: { value: 'AWS_CODECOMMIT', iconName: 'service-aws-code-deploy' },
  CUSTOM: { value: 'CUSTOM', iconName: 'build' }
}

const TriggerTypeIcons = {
  SCHEDULE: 'trigger-schedule'
}
export const getTriggerIcon = ({
  type,
  webhookSourceRepo
}: {
  type: string
  webhookSourceRepo: GetActionsListQueryParams['sourceRepo'] | undefined | string // string temporary until backend
}): IconName => {
  const webhookSourceRepoIconName =
    webhookSourceRepo && GitSourceProviders[webhookSourceRepo as GetActionsListQueryParams['sourceRepo']]?.iconName
  if (type === TriggerTypes.WEBHOOK && webhookSourceRepoIconName) {
    return webhookSourceRepoIconName as IconName
  } else if (type === TriggerTypes.SCHEDULE) {
    return TriggerTypeIcons.SCHEDULE as IconName
  }
  // placeholder for now
  return GitSourceProviders.GITHUB?.iconName as IconName
}

const triggerDrawerMap = (getString: (key: StringKeys) => string): AddDrawerMapInterface => ({
  drawerLabel: getString('pipeline-triggers.triggersLabel'),
  drawerSubLabel: getString('pipeline-triggers.triggersSubLabel'),
  showAllLabel: getString('pipeline-triggers.showAllTriggers'),
  searchPlaceholder: getString('pipeline-triggers.searchPlaceholder'),
  categories: [
    // {
    //   categoryLabel: getString('pipeline-triggers.onNewArtifactTitle'),
    //   categoryValue: 'OnArtifact',
    //   items: [
    //     {
    //       itemLabel: getString('pipeline-triggers.newArtifactLabel'),
    //       value: 'NewArtifact',
    //       iconName: 'trigger-artifact' as IconName
    //     },
    //     {
    //       itemLabel: getString('pipeline-triggers.newManifestLabel'),
    //       value: 'NewManifest',
    //       iconName: 'trigger-artifact' as IconName
    //     }
    //   ]
    // },
    {
      categoryLabel: getString('pipeline-triggers.scheduledLabel'),
      categoryValue: 'Scheduled',
      items: [
        {
          itemLabel: getString('pipeline-triggers.cronLabel'),
          value: 'Cron',
          iconName: TriggerTypeIcons.SCHEDULE as IconName
        }
      ]
    },
    {
      categoryLabel: getString('execution.triggerType.WEBHOOK'),
      categoryValue: 'Webhook',
      items: [
        {
          itemLabel: getString('common.repo_provider.githubLabel'),
          value: GitSourceProviders.GITHUB.value,
          iconName: GitSourceProviders.GITHUB.iconName
        },
        {
          itemLabel: getString('common.repo_provider.gitlabLabel'),
          value: GitSourceProviders.GITLAB.value,
          iconName: GitSourceProviders.GITLAB.iconName
        },
        {
          itemLabel: getString('common.repo_provider.bitbucketLabel'),
          value: GitSourceProviders.BITBUCKET.value,
          iconName: GitSourceProviders.BITBUCKET.iconName
        },
        {
          itemLabel: getString('common.repo_provider.codecommit'),
          value: GitSourceProviders.AWS_CODECOMMIT.value,
          iconName: GitSourceProviders.AWS_CODECOMMIT.iconName
        },
        {
          itemLabel: getString('common.repo_provider.customLabel'),
          value: GitSourceProviders.CUSTOM.value,
          iconName: GitSourceProviders.CUSTOM.iconName
        }
      ]
    }
  ]
})

export const getSourceRepoOptions = (getString: (str: StringKeys) => string): { label: string; value: string }[] => [
  { label: getString('common.repo_provider.githubLabel'), value: GitSourceProviders.GITHUB.value },
  { label: getString('common.repo_provider.gitlabLabel'), value: GitSourceProviders.GITLAB.value },
  { label: getString('common.repo_provider.bitbucketLabel'), value: GitSourceProviders.BITBUCKET.value },
  { label: getString('common.repo_provider.codecommit'), value: GitSourceProviders.AWS_CODECOMMIT.value },
  { label: getString('common.repo_provider.customLabel'), value: GitSourceProviders.CUSTOM.value }
]

export const getCategoryItems = (getString: (key: StringKeys) => string): AddDrawerMapInterface =>
  triggerDrawerMap(getString)

export interface ItemInterface {
  itemLabel: string
  iconName: IconName
  value: string
  visible?: boolean
  disabled?: boolean
  categoryValue?: string
}

export interface TriggerDataInterface {
  triggerType: string
  sourceRepo?: string
  // all else optional
}
