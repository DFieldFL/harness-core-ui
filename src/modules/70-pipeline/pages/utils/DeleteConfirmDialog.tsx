import React, { useState } from 'react'
import { defaultTo, pick } from 'lodash-es'
import { useParams } from 'react-router-dom'
import { Intent, TextArea } from '@blueprintjs/core'
import { Text, useConfirmationDialog, useToaster } from '@wings-software/uicore'
import { EntityGitDetails, useDeleteInputSetForPipeline, useSoftDeletePipeline } from 'services/pipeline-ng'
import { useStrings } from 'framework/strings'
import { useDeleteTemplateVersionsOfIdentifier } from 'services/template-ng'
import type { ResponseBoolean } from 'services/cd-ng'
import type { PipelineDTO } from '../pipelines/views/PipelineListView'
import type { InputSetLocal } from '../inputSet-list/InputSetListView'

const useGetEntityText = (entityType: string): string => {
  const { getString } = useStrings()

  switch (entityType) {
    case 'pipeline':
      return getString('common.pipeline')
    case 'inputSet':
      return getString('inputSets.inputSetLabel')
    case 'overlayInputSet':
      return getString('inputSets.overlayInputSet')
    default:
      return ''
  }
}
interface DeleteConfirmDialogContentProps {
  gitDetails?: EntityGitDetails
  entityName: string
  commitMsg: string
  onCommitMsgChange: (commitMsg: string) => void
  entityType: string
}

export const DeleteConfirmDialogContent: React.FC<DeleteConfirmDialogContentProps> = ({
  gitDetails = {},
  entityName = '',
  commitMsg,
  onCommitMsgChange,
  entityType = ''
}): JSX.Element => {
  const { getString } = useStrings()
  return (
    <div className={'pipelineDeleteDialog'}>
      <Text margin={{ bottom: 'medium' }} title={entityName}>{`${getString(
        'common.git.confirmDelete'
      )} ${useGetEntityText(entityType)} ${entityName}?`}</Text>
      {gitDetails?.objectId && (
        <>
          <Text>{getString('common.git.commitMessage')}</Text>
          <TextArea
            value={commitMsg}
            onInput={(event: React.ChangeEvent<HTMLTextAreaElement>) => {
              onCommitMsgChange(event.target.value)
            }}
          />
        </>
      )}
    </div>
  )
}

interface DeleteMetaData {
  versions?: string[]
}

export default function useDeleteConfirmationDialog(
  entityData: PipelineDTO | InputSetLocal,
  entityType: string,
  refetchData?: () => void,
  pipelineIdentifier = '',
  setLoading?: (isLoading: boolean) => void
): { confirmDelete: (deleteMetaData?: DeleteMetaData) => void } {
  const { showSuccess, showError } = useToaster()
  const { getString } = useStrings()

  const { projectIdentifier, orgIdentifier, accountId } = useParams<{
    projectIdentifier: string
    orgIdentifier: string
    accountId: string
  }>()

  const [deleteCallMetaData, setDeleteCallMetaData] = useState<DeleteMetaData>()

  const getSuccessMessage = (): string => {
    switch (entityType) {
      case 'pipeline':
        return getString('pipeline-list.pipelineDeleted', { name: entityData.name })
      case 'inputSet':
      case 'overlayInputSet':
        return getString('inputSets.inputSetDeleted', { name: entityData.name })
      case 'template':
        return getString('common.template.deleteTemplate.templatesDeleted', { name: entityData.name })
      default:
        return ''
    }
  }

  const getErrorMessage = (): string => {
    switch (entityType) {
      case 'pipeline':
        return `pipeline.delete.pipeline.error`
      case 'inputSet':
      case 'overlayInputSet':
        return `pipeline.delete.inputset.error`
      case 'template':
        return `common.template.deleteTemplate.errorWhileDeleting`
      default:
        return ''
    }
  }

  const [commitMsg, setCommitMsg] = React.useState<string>(`${getString('delete')} ${entityData.name}`)

  const gitParams = entityData.gitDetails?.objectId
    ? {
        ...pick(entityData.gitDetails, ['branch', 'repoIdentifier', 'filePath', 'rootFolder']),
        commitMsg,
        lastObjectId: entityData.gitDetails?.objectId
      }
    : {}

  const { mutate: deletePipeline } = useSoftDeletePipeline({
    queryParams: {
      accountIdentifier: accountId,
      orgIdentifier,
      projectIdentifier,
      ...gitParams
    }
  })

  const { mutate: deleteInputSet } = useDeleteInputSetForPipeline({
    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier, pipelineIdentifier, ...gitParams }
  })

  const { mutate: deleteTemplates } = useDeleteTemplateVersionsOfIdentifier({})

  const deleteEntity = async (): Promise<ResponseBoolean> => {
    const contentType = 'application/json'
    let deleted = null
    if (entityType === 'pipeline') {
      deleted = await deletePipeline(entityData.identifier || /* istanbul ignore next */ '', {
        headers: { 'content-type': contentType }
      })
    } else if (entityType === 'template') {
      deleted = await deleteTemplates(defaultTo(entityData?.identifier, ''), {
        queryParams: {
          accountIdentifier: accountId,
          orgIdentifier,
          projectIdentifier,
          ...gitParams
        },
        body: JSON.stringify({ templateVersionLabels: deleteCallMetaData?.versions }),
        headers: { 'content-type': contentType }
      })
    } else {
      deleted = await deleteInputSet(entityData.identifier || /* istanbul ignore next */ '', {
        headers: { 'content-type': contentType }
      })
    }
    return deleted
  }

  const { openDialog: openConfirmDeleteDialog } = useConfirmationDialog({
    contentText: (
      <DeleteConfirmDialogContent
        entityName={entityData?.name || ''}
        entityType={entityType}
        gitDetails={entityData.gitDetails}
        commitMsg={commitMsg}
        onCommitMsgChange={setCommitMsg}
      />
    ),
    titleText: `${getString('delete')} ${useGetEntityText(entityType)}`,
    confirmButtonText: getString('delete'),
    cancelButtonText: getString('cancel'),
    intent: Intent.DANGER,
    buttonIntent: Intent.DANGER,
    onCloseDialog: async (isConfirmed: boolean) => {
      /* istanbul ignore else */
      if (isConfirmed) {
        try {
          setLoading?.(true)
          const deleted = await deleteEntity()
          setLoading?.(false)
          /* istanbul ignore else */
          if (deleted?.status === 'SUCCESS') {
            showSuccess(getSuccessMessage())
          } else {
            throw getString('somethingWentWrong')
          }
          refetchData?.()
        } catch (err) {
          /* istanbul ignore next */
          showError(err?.data?.message || err?.message, undefined, getErrorMessage())
        }
      }
    }
  })

  const confirmDelete = (deleteMetaData?: DeleteMetaData): void => {
    setDeleteCallMetaData(deleteMetaData)
    openConfirmDeleteDialog()
  }
  return { confirmDelete }
}
