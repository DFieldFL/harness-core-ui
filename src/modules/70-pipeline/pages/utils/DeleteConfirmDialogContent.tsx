import React from 'react'
import { TextArea } from '@blueprintjs/core'
import { Text } from '@wings-software/uicore'
import type { EntityGitDetails } from 'services/pipeline-ng'
import { useStrings } from 'framework/strings'

interface DeleteConfirmDialogContentProps {
  gitDetails?: EntityGitDetails
  entityName: string
  commitMsg: string
  onCommitMsgChange: (commitMsg: string) => void
  entityType: string
}

export const DeleteConfirmDialogContent: React.FC<DeleteConfirmDialogContentProps> = ({
  gitDetails = {},
  entityName = '',
  commitMsg,
  onCommitMsgChange,
  entityType = ''
}): JSX.Element => {
  const { getString } = useStrings()

  return (
    <div className={'pipelineDeleteDialog'}>
      <Text margin={{ bottom: 'medium' }} title={entityName}>{`${getString(
        'pipeline-list.confirmDelete'
      )} ${entityType} ${entityName}?`}</Text>
      {gitDetails?.objectId && (
        <>
          <Text>{getString('common.git.commitMessage')}</Text>
          <TextArea
            value={commitMsg}
            onInput={(event: React.ChangeEvent<HTMLTextAreaElement>) => {
              onCommitMsgChange(event.target.value)
            }}
          />
        </>
      )}
    </div>
  )
}
