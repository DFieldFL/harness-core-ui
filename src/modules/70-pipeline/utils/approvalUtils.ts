import type { ApprovalInstanceResponse } from 'services/pipeline-ng'

export type ApprovalStatusType = ApprovalInstanceResponse['status']

export const ApprovalStatus: Record<ApprovalStatusType, ApprovalStatusType> = {
  WAITING: 'WAITING',
  APPROVED: 'APPROVED',
  REJECTED: 'REJECTED',
  FAILED: 'FAILED',
  EXPIRED: 'EXPIRED'
}

export function isApprovalWaiting(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.WAITING
}

export function isApprovalApproved(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.APPROVED
}

export function isApprovalRejected(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.REJECTED
}

export function isApprovalFailed(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.FAILED
}

export function isApprovalExpired(status: ApprovalStatusType): boolean {
  return status === ApprovalStatus.EXPIRED
}
