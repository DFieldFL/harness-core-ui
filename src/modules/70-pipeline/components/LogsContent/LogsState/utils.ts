import { escapeStringRegexp } from '@common/utils/StringUtils'

import type { State } from './types'

export function getRegexForSearch(str: string): RegExp {
  return new RegExp(escapeStringRegexp(str), 'gi')
}

export function getDefaultReducerState(props: Partial<Pick<State, 'selectedStage' | 'selectedStep'>> = {}): State {
  return {
    units: [],
    logKeys: [],
    selectedStage: props.selectedStage || '',
    selectedStep: props.selectedStep || '',
    dataMap: {},
    searchData: {
      text: '',
      currentIndex: 0,
      linesWithResults: []
    }
  }
}

export const logsCache = new Map<string, string>()
