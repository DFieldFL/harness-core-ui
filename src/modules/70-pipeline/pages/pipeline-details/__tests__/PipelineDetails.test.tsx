import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { defaultAppStoreValues } from '@projects-orgs/pages/projects/__tests__/DefaultAppStoreData'
import routes from '@common/RouteDefinitions'
import { accountPathProps, pipelineModuleParams, pipelinePathProps } from '@common/utils/routeUtils'
import PipelineDetails from '../PipelineDetails'
import { PipelineResponse } from './PipelineDetailsMocks'
jest.mock('services/pipeline-ng', () => ({
  useGetPipelineSummary: jest.fn(() => PipelineResponse)
}))

const TEST_PATH = routes.toPipelineDetail({ ...accountPathProps, ...pipelinePathProps, ...pipelineModuleParams })
describe('Pipeline Details tests', () => {
  test('render snapshot view', async () => {
    const { container } = render(
      <TestWrapper
        path={TEST_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <PipelineDetails />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
