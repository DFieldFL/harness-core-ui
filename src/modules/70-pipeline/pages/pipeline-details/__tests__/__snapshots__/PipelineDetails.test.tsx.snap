// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipeline Details tests render snapshot view 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main container standard StyledProps--flex StyledProps--padding-left-large StyledProps--padding-right-large"
  >
    <div
      class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
    >
      <div
        class="breadcrumbs"
      >
        <div>
          <a
            class="breadcrumb"
            href="/account/testAcc/cd/orgs/testOrg/projects/test/dashboard"
          >
            test
          </a>
          <span
            class="bp3-icon StyledProps--main separator"
            data-icon="main-chevron-right"
          >
            <svg
              height="11"
              viewBox="0 0 8 14"
              width="11"
            >
              <path
                d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                fill="#4d5969"
              />
            </svg>
          </span>
        </div>
        <div>
          <a
            class="breadcrumb"
            href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines"
          >
            pipelineBreadcrumb
          </a>
          <span
            class="bp3-icon StyledProps--main separator"
            data-icon="main-chevron-right"
          >
            <svg
              height="11"
              viewBox="0 0 8 14"
              width="11"
            >
              <path
                d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                fill="#4d5969"
              />
            </svg>
          </span>
        </div>
        <div>
          <span
            class="lastBreadcrumb"
          >
            testsdfsdf
          </span>
        </div>
      </div>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main TabNavigation--container"
    >
      <a
        class="TabNavigation--tags TabNavigation--small"
        href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines/pipeline/pipeline-studio/"
      >
        pipelineStudio
      </a>
      <a
        class="TabNavigation--tags TabNavigation--small"
        href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines/pipeline/input-sets"
      >
        inputSetsText
      </a>
      <a
        class="TabNavigation--tags TabNavigation--small"
        href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines/pipeline/triggers"
      >
        pipeline.triggers.triggersLabel
      </a>
      <a
        class="TabNavigation--tags TabNavigation--small"
        href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines/pipeline/executions"
      >
        executionHeaderText
      </a>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main pageBody"
    data-name="page-body"
  />
</div>
`;
